mountinfo/mountinfo_unix.go

```diff
@@ -0,0 +1,37 @@
+// +build !windows
+
+package mountinfo
+
+import (
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+)
+
+// Mounted determines if path is mounted (i.e. it is a mount point).
+func Mounted(path string) (bool, error) {
+	// Try the fast method (of comparing devices) first.
+	// Alas, it can't detect bind mounts, so fall back on
+	// good ol' mountinfo parsing.
+	var st, pst unix.Stat_t
+
+	err := unix.Lstat(path, &st)
+	if err != nil {
+		return false, err
+	}
+	err = unix.Lstat(filepath.Dir(path), &pst)
+	if err != nil {
```

Comment:
The code that calls this usually does mount/unmount next, and for that, root is required.

And yet I was thinking that maybe instead of returning an error we can just jump to the fallback code of parsing mountinfo.

Reply:
In fact you were right. If path does not exist, this means it is not mounted (since a mount point can not be removed).

In all the other cases I fall back to mountinfo parsing